drivers/interrupt_controller/intc_ioapic.c

```diff
@@ -314,35 +310,21 @@ int ioapic_resume_from_suspend(const struct device *port)
 */
 __pinned_func
 static int ioapic_device_ctrl(const struct device *dev,
-			      uint32_t ctrl_command,
-			      enum pm_device_state *state)
+			      enum pm_device_state state)
 {
 	int ret = 0;
 
-	if (ctrl_command == PM_DEVICE_STATE_SET) {
-		switch (*state) {
-		case PM_DEVICE_STATE_LOW_POWER:
-			break;
-		case PM_DEVICE_STATE_ACTIVE:
-			if (ioapic_device_power_state !=
-					PM_DEVICE_STATE_LOW_POWER) {
-				ret = ioapic_resume_from_suspend(dev);
-			}
-			break;
-		case PM_DEVICE_STATE_SUSPEND:
-		case PM_DEVICE_STATE_FORCE_SUSPEND:
-		case PM_DEVICE_STATE_OFF:
-			ret = ioapic_suspend(dev);
-			break;
-		default:
-			ret = -ENOTSUP;
-		}
-
-		if (ret == 0) {
-			ioapic_device_power_state = *state;
-		}
-	} else if (ctrl_command == PM_DEVICE_STATE_GET) {
-		*state = ioapic_device_power_state;
+	switch (state) {
+	case PM_DEVICE_STATE_ACTIVE:
+		ret = ioapic_resume_from_suspend(dev);
+		break;
+	case PM_DEVICE_STATE_SUSPEND:
+	case PM_DEVICE_STATE_FORCE_SUSPEND:
+	case PM_DEVICE_STATE_OFF:
+		ret = ioapic_suspend(dev);
+		break;
+	default:
+		ret = -ENOTSUP;
```

Comment:
Original code is handling `PM_DEVICE_STATE_LOW_POWER`. Here you are not doing it and it will return `-ENOTSUP` instead of `0`.

Reply:
It's not handling it really, it does nothing. 
```
		case PM_DEVICE_STATE_LOW_POWER:
			break;
```

I can leave as it is and fix this problem later. I don't think PM subsys is handling `-ENOTSUP` correctly as of today.

Reply:
added the case back, will fix these sort of issues later.